vm.go

```diff
@@ -362,7 +363,7 @@ func (vm *VirtualMachineSpec) HotDetachEndpoint(endpoint *hcn.HostComputeEndpoin
 	// Hot detach an endpoint from the compute system
 	request := hcsschema.ModifySettingRequest{
 		RequestType:  requesttype.Remove,
-		ResourcePath: path.Join("VirtualMachine/Devices/NetworkAdapters", endpoint.Id),
```

Comment:
what's the original motivation to change it from `Id` to `Name`?

Reply:
If we use the ID, the ID changes every time the VM is rebooted. This is a problem for the Windows VM because it detects a different physical network adapter and loses its networking configuration.

I.e. if we use the ID, a different physical network adapter is mapped into the VM each time the VM restarts.

Reply:
Add an optional input parameter to allow caller specify resource name, if caller specify resource name, use it, otherwise use ID to ensure no conflict to existing resource name.

Same for hotAttachEndpoint